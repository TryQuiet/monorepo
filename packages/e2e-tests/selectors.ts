import { Selector, t } from 'testcafe'

export class LoadingPanel {
  private readonly text: string

  constructor(title: string) {
    this.text = title
  }

  get title() {
    return Selector('span').withText(this.text)
  }
}

export class Channel {
  private readonly name: string

  constructor(name: string) {
    this.name = name
  }

  get title() {
    return Selector('h6').withText(`#${this.name}`)
  }

  get messagesList() {
    return Selector('ul').withAttribute('id', 'messages-scroll')
  }

  get messagesGroup() {
    return this.messagesList.find('li')
  }

  get messagesGroupContent() {
    return this.messagesGroup.find('p').withAttribute('data-testid', /messagesGroupContent-/)
  }

  get messageInput () {
    return Selector('div').withAttribute('data-testid', 'messageInput')
  }

  async sendMessage(message: string) {
    await t.typeText(this.messageInput, message)
    await t.pressKey('enter')
  }
}
