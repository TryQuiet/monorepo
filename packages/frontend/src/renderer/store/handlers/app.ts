import { produce, immerable } from 'immer'
import { createAction, handleActions } from 'redux-actions'
import { actionTypes } from '../../../shared/static'

import { ActionsType, PayloadType } from './types'

export class App {
  version: number

  constructor(values?: Partial<App>) {
    Object.assign(this, values)
    this[immerable] = true
  }
}

export const initialState: App = {
  ...new App({
    version: null
  })
}

const loadVersion = createAction(actionTypes.SET_APP_VERSION, () => 1.0)

export const actions = {
  loadVersion
}

export type AppActions = ActionsType<typeof actions>

export const reducer = handleActions<App, PayloadType<AppActions>>(
  {
    [loadVersion.toString()]: (state, { payload: version }: AppActions['loadVersion']) =>
      produce(state, draft => {
        draft.version = version
      })
  },
  initialState
)

export default {
  actions,
  reducer
}
