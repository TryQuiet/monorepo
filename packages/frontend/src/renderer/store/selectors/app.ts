import { createSelector } from 'reselect'

import { Store } from '../reducers'

const app = (s: Store) => s.app

const version = createSelector(app, (a) => a.version)

export default {
  version
}
