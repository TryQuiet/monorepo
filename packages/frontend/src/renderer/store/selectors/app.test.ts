/* eslint import/first: 0 */
import selectors from './app'
import { initialState as AppState } from '../handlers/app'

import create from '../create'

describe('app -', () => {
  let store = null
  beforeEach(() => {
    jest.clearAllMocks()
    store = create({
      app: {
        ...AppState,
        version: '0.13.37',
        transfers: {},
        modalTabToOpen: 'addFunds',
        allTransfersCount: 12,
        newTransfersCounter: 2
      }
    })
  })

  it('version selector', async () => {
    expect(selectors.version(store.getState())).toMatchSnapshot()
  })
})
