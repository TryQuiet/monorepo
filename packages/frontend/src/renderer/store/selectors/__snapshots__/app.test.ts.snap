// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app - version selector 1`] = `"0.13.37"`;
